File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (head == nullptr) {
            return nullptr;
        }
        ListNode* pointer = head->next;
        ListNode* pNode = head;
        while (pointer) { 
            if (pointer->val == pNode->val) {
                pNode->next = pointer->next;
                ListNode* temp = pointer->next;
                delete pointer;
                pointer = temp;
            } else {
                pNode = pointer;
                pointer = pointer->next;
            }
        }
        return head;
    }
};

File: 83-remove-duplicates-from-sorted-list/83-remove-duplicates-from-sorted-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    if (head == null) {
        return null;
    }
    let pointer = head.next;
    let pNode = head;
    while (pointer) { 
        if (pointer.val == pNode.val) {
            pNode.next = pointer.next;
            let temp = pointer.next;
            delete pointer;
            pointer = temp;
        } else {
            pNode = pointer;
            pointer = pointer.next;
        }
    }
    return head;
};